src/Kupo/App/Http.hs

```diff
@@ -170,8 +172,10 @@ app withDatabase patternsVar readHealth req send =
     routePatterns = \case
         ("GET", []) ->
             readTVarIO patternsVar >>= send . handleGetPatterns
-        ("GET", _) ->
-            send Errors.notFound
+        ("GET", args) ->
+            withDatabase (send .
+                handleGetMatchingPatterns (patternFromPath args)
```

Comment:
Note that the patterns are 'cached' in-memory in the `patternsVar :: TVar IO [Pattern]`. Thus, for read-only manipulation like this one, you don't  need a database connection (see above `readTVarIO patternsVar >>=`).